pkg/asset/installconfig/gcp/client.go

```diff
@@ -222,6 +225,34 @@ func (c *Client) GetProjects(ctx context.Context) (map[string]string, error) {
 	return projects, nil
 }
 
+// GetRegions gets the regions that are valid for the project. An error is returned when unsuccessful
+func (c *Client) GetRegions(ctx context.Context, project string) (map[string]string, error) {
+	computeCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	svc, err := c.getComputeService(computeCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	regionCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	gcpRegionsList, err := svc.Regions.List(project).Context(regionCtx).Do()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get regions for project %s", project)
+	}
+
+	validatedRegions := make(map[string]string)
+	for _, region := range gcpRegionsList.Items {
+		// Only add validated regions
+		if value, ok := gcpValidation.Regions[region.Name]; ok {
+			validatedRegions[region.Name] = value
+		}
+	}
```

Comment:
I don't *think* this is where we want to filter the list of regions from the GCP API down to the list whitelisted by QE.  I think this function should just return the complete list from the API, and that should be used to validate the region passed in the install config.

Reply:
Is it a waste of resources to convert it to a map? I thought that we decided to pass around the map. If/when we get the regions it will return a list. If we don't filter it here (and we want to return a map) it will be filled with empty values.

Reply:
My main concern is the functional correctness, so let's go into that first. The behavior we want is:

- the survey is populated by the subset of regions from the GCP API that appear in the "qe whitelist" in `gcpValidation.Regions`
- the install config will accept any region from the GCP API

The example to test this we came up with is `europe-west8`. `europe-west8` (at the moment) should not appear in the survey, but should be a valid region when passed in the install config. I believe if we filter the region list to the validated QE list here, the second scenario will not work. 

 >  If we don't filter it here (and we want to return a map) it will be filled with empty values

IMHO this should be filled with data from the API, not the static list from our code. But I am happy to go with a different preferred solution as long as the functionality works in the desired way.